### PR TITLE
JSON Schema validation via TestNG

### PR DESCRIPTION
**Pull request by alszla**

### CODE REVIEW

File: src/main/java/com/foundation/salesforce/core/utils/ResponseValidation.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+        InputStream inputStream = null;
+        try  {
+            inputStream = getClass().getClassLoader().getResourceAsStream(schemaTypeName.concat(".json"));
+            JSONObject rawSchema = new JSONObject(new JSONTokener(inputStream));
+            Schema schema = SchemaLoader.load(rawSchema);
+            schema.validate(new JSONObject(response.jsonPath().getMap("$")));
+            return true;
+        } catch (ValidationException npvex) {
+            //Assert.fail("Мне похуй!");
```
Comment (MelissaRoman06) [inline src/main/java/com/foundation/salesforce/core/utils/ResponseValidation.java]: Please delete comments.

Comment (alszla) [reply src/main/java/com/foundation/salesforce/core/utils/ResponseValidation.java]: He, he, debugging leftovers, sorry. I'll take'em out ASAP.

Comment (alszla) [reply src/main/java/com/foundation/salesforce/core/utils/ResponseValidation.java]: Done.

**Review by MelissaRoman06 — APPROVED** [review]

File: src/test/java/com/foundation/salesforce/steps/TaskAcceptanceSteps.java [hunk 2]
```diff
@@ -37,77 +38,90 @@
     private ValidatableResponse json;
     private Response response;
     private TaskApi taskApi;
-
-    Task task = new Task();
+    private Task task;
 
     /**
-     * Retrieves an authentication token in order to be able to access to Salesforce platform.
+     * Class constructor.
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/TaskAcceptanceSteps.java:44]: Please be a little more specific in this comment.

File: src/test/java/com/foundation/salesforce/steps/TaskAcceptanceSteps.java [hunk 3]
```diff
@@ -37,77 +38,90 @@
...
      */
-    @Given("a user logs in into the Task page")
-    public void a_user_logs_in() {
+    public TaskAcceptanceSteps(Task task) {
+        this.task = task;
         taskApi = TaskApi.getInstance();
     }
 
     /**
      *
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/TaskAcceptanceSteps.java:61]: Why in this part don't have comment.

File: src/test/java/com/foundation/salesforce/steps/TaskAcceptanceSteps.java (hunk 2 above)
Comment (alszla) [reply src/test/java/com/foundation/salesforce/steps/TaskAcceptanceSteps.java:44]: OK, I'll be more specific in this section. Promise.

Comment (alszla) [reply src/test/java/com/foundation/salesforce/steps/TaskAcceptanceSteps.java:44]: Done.

File: src/test/java/com/foundation/salesforce/steps/TaskAcceptanceSteps.java (hunk 3 above)
Comment (alszla) [reply src/test/java/com/foundation/salesforce/steps/TaskAcceptanceSteps.java:61]: Done.

**Review by titor-oopart — APPROVED** [review]

File: src/main/java/com/foundation/salesforce/core/utils/ResponseValidation.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    public static ResponseValidation getInstance() {
+        if (responseValidation == null) {
+            responseValidation = new ResponseValidation();
+        }
+        return responseValidation;
+    }
+
+    /**
+     *
```
Comment (Cxristian) [inline src/main/java/com/foundation/salesforce/core/utils/ResponseValidation.java:53]: the name of the method is missing...

File: src/test/resources/features/TaskAcceptance.feature [hunk 5]
```diff
@@ -1,30 +1,52 @@
...
-    Then status code is 200 after deleting
+  @UpdateTask
+  Scenario: User updates an existing task
+    Given user specifies body content
+      | Status   | Deferred    |
+      | Priority | High        |
+    When user patches an existing task
+    Then status code is 204
+
+#  Scenario: User updates a task by id and specifying at least a status and priority
```
Comment (Cxristian) [inline src/test/resources/features/TaskAcceptance.feature:21]: remove commented lines

**Review by TheYisusGithub — APPROVED** [review]

File: src/test/resources/features/TaskAcceptance.feature (hunk 5 above)
Comment (alszla) [reply src/test/resources/features/TaskAcceptance.feature:21]: Fixed.

File: src/main/java/com/foundation/salesforce/core/utils/ResponseValidation.java (hunk 4 above)
Comment (alszla) [reply src/main/java/com/foundation/salesforce/core/utils/ResponseValidation.java:53]: Fixed.

**Pull request merged**